src/js/media-chrome-listbox.js

```diff
@@ -0,0 +1,266 @@
+import { MediaUIAttributes } from './constants.js';
+import { defineCustomElement } from './utils/defineCustomElement.js';
+import {
+  Window as window,
+  Document as document,
+} from './utils/server-safe-globals.js';
+
+const template = document.createElement('template');
+
+template.innerHTML = `
+<style>
+  :host ul {
+    list-style: none;
+  }
+  ::slotted([tabindex="0"]) {
+    background-color: lightgrey;
+  }
+</style>
+<ul tabindex="0" role="listbox">
+  <slot></slot>
+</ul>
+`;
+
+class MediaChromeListbox extends window.HTMLElement {
+  #keysSoFar = '';
+  #clearKeysTimeout = null;
+
+  static get observedAttributes() {
+    return ['disabled', MediaUIAttributes.MEDIA_CONTROLLER];
+  }
+
+  constructor(options = {}) {
+    super();
+
+    const shadow = this.attachShadow({ mode: 'open' });
+
+    const listboxHTML = template.content.cloneNode(true);
+    this.nativeEl = listboxHTML;
+
+    shadow.appendChild(listboxHTML);
+
+    const slot = this.shadowRoot.querySelector('slot');
```

Comment:
nit (nb): might wanna cache this query, `this.shadowRoot.querySelector('slot')` is used 8 times in this class